src/get-started/java-binding.md

```diff
@@ -0,0 +1,382 @@
+to its state:
+
+??? note "MySchemaTest"
+    ```java
+    package com.exonum.binding.qaservice;
+
+    import com.exonum.binding.proxy.Cleaner;
+    import com.exonum.binding.storage.database.Fork;
+    import com.exonum.binding.storage.database.MemoryDb;
+    import com.exonum.binding.storage.database.Snapshot;
+    import org.junit.After;
+    import org.junit.Before;
+    import org.junit.Test;
+
+    public class MySchemaTest {
+
+      MemoryDb db;
+      Cleaner cleaner;
+
+      @Before
+      public void setUp() {
+        db = MemoryDb.newInstance();
+        cleaner = new Cleaner();
+      }
+
+      @After
+      public void tearDown() throws Exception {
+        cleaner.close();
+        db.close();
+      }
+
+      @Test
+      public void testCreatingSnapshot() {
+        Snapshot snapshot = db.createSnapshot(cleaner);
+        // Use the Snapshot to test the schema
+      }
+
+      @Test
+      public void testMergingSomeChanges() {
+        Fork fork = db.createFork(cleaner);
+        // Make some changes to Fork
+        // …
+
+        // Merge the changes made to the database Fork into the database
+        db.merge(fork);
+
+        // Create a Snapshot of the database state. Must include the changes
+        // made above.
+        Snapshot snapshot = db.createSnapshot(cleaner);
+        // Use the Snapshot to test the schema/transaction
+        // with a certain database state.
+        // …
+        }
+      }
+      ```
+
+##### Of Transactions
+
+To test transactions execution, you can use `MemoryDb`, as in the previous
+section.
+
+#### Of Read Requests
+
+To test read requests for service data, you can use a fake that implements
+`Node` interface and uses `MemoryDb` to create `Snapshot`:
+`com.exonum.binding.service.NodeFake`. The `MemoryDb` contents can be filled in
+by executing `MemoryDb#merge(Fork)` operation as in the section above.
+
+#### Of API
+
+To test API implemented by Vertx tools, use the tools described in the
+[project documentation](https://vertx.io/docs/vertx-unit/java/#_introduction)
+
+You can use [Vertx Web Client][vertx-web-client] as a client or another HTTP
+client.
+
+An example of API service tests can be found in
+`com.exonum.binding.cryptocurrency.ApiControllerTest`.
+
+### Known Limitations
+
+- Serialization is determined by a user, so Java services are not compatible
+  with JS light client (unless messages are updated).
+- Core collections necessary to form a complete cryptographic proof for user
+  service data (collections and their elements) are available only in a "raw"
+  form - without deserialization of the content, which makes their use somewhat
+  difficult.
+- Not all system Rust services are available (EJB App configuration, Exonum
+  time and anchoring services will be integrated).
+- Custom Rust services can be added to the application only by modifying and
+  rebuilding thereof.
+- The application supports only one Java service. However, there will be more!
```

Comment:
> However, there will be more!

Kind of awkward; I'd rephrase as "Support of multiple Java services is coming in the near future."